pyproject.toml

```diff
@@ -83,17 +83,65 @@ addopts = [
 ]
 xfail_strict = true
 
-[tool.black]
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+# Same as Black.
 line-length = 120
-target-version = ['py311']
+indent-width = 4
+show-fixes = true
+
+# Assume Python 3.12
+target-version = "py312"
```

Comment:
this should still be `py311` because we haven't dropped python3.11 support yet. that is expected to happen in January (https://power-grid-model.readthedocs.io/en/stable/release_and_support/RELEASE.html#actively-supported-python-versions)